### PR TITLE
DS-263 | @rebeccahongsf | Membership Phase 2

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Membership Phase 2 Enhancements
- Payment Update Form
- Payment History Form

# Review By (Date)
- Monday, May 13 (Launch is Wednesday, May 15)

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Pull down fresh `.env` vars from vault
3. Enable the mockserver by add `MEGAPROFILE_MOCK='true'` to your .env
4. Navigate to `/membership/payment-update`
5. Verify that a message appears and the links work as expected
6. Navigate to `/membership/payment-history`
7. Verify that the form appears with the schedule ID
8. Change the `amountDue` value in `src/utilities/mocks/memberships.js` from 0 to 600
9. Navigate to `/membership/payment-update`
10. Verify that the payment update form appears
11. Navigate to `/membership/payment-history`
12. Verify that the form appears with no schedule ID
13. Comment out the SAA membership in `src/utilities/mocks/memberships.js`
14. Navigate to `/membership/payment-history` 
15. Verify that an `Interest in joining SAA` message appears
16. Navigate to `/membership/payment-update`
17. Verify that an `Interest in joining SAA` message appears
18. Review code 👾 

# Associated Issues and/or People
- DS-170

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -47,7 +47,7 @@
     endpoint = 'https://vault.stanford.edu'
     # Order doesn't matter here as these will be fetched asyncronously.
     # Try not to have dupes.
-    secrets = ['secret/data/projects/adapt/saml', 'secret/data/projects/adapt/alumni']
+    secrets = ['secret/data/projects/adapt/saml', 'secret/data/projects/adapt/membership']
```
Comment (sherakama) [inline netlify.toml:50]: TODO - Change before merge.

**Comment by yvonnetangsu** [comment]
@sherakama I saw that you tagged me for review - is this to set up a preview for membership changes?

**Comment by sherakama** [comment]
> @sherakama I saw that you tagged me for review - is this to set up a preview for membership changes?

Oh I totally meant to leave a comment too. I flagged you as I moved this work here and you mentioned the other day that you were watching some of Rebecca's stuff. Nothing to do until the new year.

**Comment by yvonnetangsu** [comment]
> > @sherakama I saw that you tagged me for review - is this to set up a preview for membership changes?
> 
> Oh I totally meant to leave a comment too. I flagged you as I moved this work here and you mentioned the other day that you were watching some of Rebecca's stuff. Nothing to do until the new year.

Thanks! Glad to wait till next year 😄

File: src/styles/light-forms.css [hunk 2]
```diff
@@ -1,356 +1,360 @@
...
+.su-light-form .ggeWidget .ggeQuestion.ggeQuestion--readonly.su-name label {
   @apply su-hidden;
 }
 
-.ggeWidget[class*="membership-forms"] .ggeSection--askArray.ggeSection--registration,
-.ggeWidget[class*="membership-forms"] .ggeTally {
-  @apply su-border-2 su-pt-26 su-px-26 su-pb-45 su-mb-72 su-border-4 su-border-digital-blue su-bg-fog-light;
+.su-light-form .ggeWidget .ggeSection--askArray.ggeSection--registration,
+.su-light-form .ggeWidget .ggeTally {
+  @apply su-border-2 su-pt-[2.6rem] su-px-[2.6rem] su-pb-[4.5rem] su-mb-[7.2rem] su-border-4 su-border-digital-blue su-bg-fog-light;
```
Comment (yvonnetangsu) [inline src/styles/light-forms.css:65]: Is there a reason we're adding square brackets to these padding/margin classes? Out of the box Decanter honors all of the values here in this list, so these ones here are all supported without the square brackets 😄 
https://github.com/SU-SWS/decanter/blob/main/src/plugins/theme/spacing.js.

Comment (rebeccahongsf) [reply src/styles/light-forms.css:65]: There was no specific reason for the square brackets 😅 Thanks for checking in, @yvonnetangsu! I'll have that updated momentarily 🚀

**Comment by yvonnetangsu** [comment]
Oh @sherakama @rebeccahongsf it seems that this PR is undoing my CSS changes in `light-form.css` in this release 👀 
https://github.com/SU-SWS/saa_alumni/pull/786/files#diff-1fd0c61852aee148aca4035256994732e6c59450145775d3d93bce8a3b1a1935

File: src/styles/light-forms.css (hunk 2 above)
Comment (yvonnetangsu) [reply src/styles/light-forms.css:65]: Thank you @rebeccahongsf ! You're probably still working on it so I'll hold off on reviewing till you're ready 😁

**Comment by rebeccahongsf** [comment]
> Oh @sherakama @rebeccahongsf it seems that this PR is undoing my CSS changes in `light-form.css` in this release 👀 https://github.com/SU-SWS/saa_alumni/pull/786/files#diff-1fd0c61852aee148aca4035256994732e6c59450145775d3d93bce8a3b1a1935

Ah! Thank you for reminding me of the changes made during the decanter updates @yvonnetangsu! I'll make sure to do a cross comparison of the changes and fix up the stylesheet in this PR 😉

**Comment by rebeccahongsf** [comment]
@sherakama @yvonnetangsu the membership work is finally ready for review! 🖖

File: src/components/page-types/lightFormPage/lightFormPage.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    </Heading>
+    <p className="su-card-paragraph">
+      <SbLink link={{ url: '/membership/join' }}>Join now</SbLink> or visit our{' '}
+      <SbLink link={{ url: '/membership/faq' }}>Membership FAQs</SbLink> page to
+      learn more details about our membership planms and benefits.
+    </p>
+    <p className="su-card-paragraph su-mb-0">
+      If you purchased your membership online more than one business day ago,
+      and your membership is not recognized after logging in, call us at (650)
+      725-0692 or email us at{' '}
```
Comment (sherakama) [inline src/components/page-types/lightFormPage/lightFormPage.js:33]:
```suggestion
      and your membership is not recognized after logging in, call us at <a href="tel:6507250692">(650)
      725-0692</a> or email us at{' '}
```

File: src/components/page-types/lightFormPage/lightFormPage.js [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+      setPaymentRefId(false);
+      setError(true);
+      setLoading(false);
+    } else if (memberships) {
+      findPaymentRefId();
+    }
+  }, [orgId, dssId, memberships, paymentRefId]);
+
+  return (
+    <AuthenticatedPage>
```
Comment (sherakama) [inline src/components/page-types/lightFormPage/lightFormPage.js:142]: You can now protect this page at the the edge. See the [netlify toml file](https://github.com/SU-SWS/saa_alumni/blob/dev/netlify.toml#L39-L57) for more info. It is good to keep the authpage component wrapping the page but also add the route to the edge function protection.

File: src/api/membership/payment/[...history].js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { GiveGabNonce } from '../../../utilities/givegab/GiveGabNonce';
+
+const paymentHandler = async (req, res) => {
```
Comment (sherakama) [inline src/api/membership/payment/[...history].js:3]: Are end users authenticated when they use this endpoint? If so, we should add auth protection on this.

**Comment by sherakama** [comment]
@rebeccahongsf or @ericbakenhus Is the failing test here a known thing? 

Screenshot 2024-05-01 at 15 57 57

**Comment by ericbakenhus** [comment]
> @rebeccahongsf or @ericbakenhus Is the failing test here a known thing?

@sherakama Yep. See: #824. Should be fixed now.

**Comment by rebeccahongsf** [comment]
@sherakama I'm looking into the cypress tests. Aside from that, are there any additional adjustments that we should make in code?

**Review by sherakama — COMMENTED** [review]
GTG on green lights. Don't forget to change back those couple of TODO comments first however.

**Comment by rebeccahongsf** [comment]
Woohoo! We have the greenlights. Thank you for the review @sherakama and thank you for the cypress fix @ericbakenhus 😃

**Review by rebeccahongsf — APPROVED** [review]
🚀

**Comment by rebeccahongsf** [comment]
How it feels to approve my own work @sherakama @yvonnetangsu 

image

**Pull request merged**